core/src/main/scala/org/http4s/rho/CompileService.scala

```diff
@@ -42,7 +42,7 @@ object CompileService {
     * @param filter [[RhoMiddleware]] to apply to the routes.
     * @return An `HttpService`
     */
-  def foldServices[F[_]: Monad](routes: Seq[RhoRoute.Tpe[F]], filter: RhoMiddleware[F]): HttpService[F] = {
+  def foldServices[F[_]: Monad](routes: Seq[RhoRoute.Tpe[F]], filter: RhoMiddleware[F]): HttpRoutes[F] = {
```

Comment:
Hmm... TBH, I think this function needs a bit more than a rename, but it might be out of the scope of this PR.  So, yes i think you should rename it to `foldRoutes`.  

If you have time, I would say remove the `filter` parameter and have the [`ServiceBuilder.toService`](https://github.com/http4s/rho/blob/v0.18.0/core/src/main/scala/org/http4s/rho/ServiceBuilder.scala#L18) function which calls `foldServices` run the middleware on the routes before passing them in.  This reduces the complexity of this function, and makes the name make more sense.